IPython/terminal/shortcuts.py

```diff
@@ -13,6 +13,7 @@
 from typing import Callable
 
 
+import parso
```

Comment:
would _definitely_ need this added to `setup.cfg#install_requires`, probably with a bottom sanity pin of `>=0.5.2`, which is the oldest parso supported by a compatible jedi... though the most recent and actually usable jedi is more `>=0.8,<0.9`. getting these wrong is _very_ painful. so it's probably worthwhile to keep the regex as a fallback until we bump to a more limiteed jedi range.

Reply:
I’ve added a regex-based fallback, how does that look?